src/ngx_http_lua_socket_udp.c

```diff
@@ -865,6 +895,7 @@ ngx_http_lua_socket_udp_send(lua_State *L)
             return luaL_error(L, "impossible to reach here");
     }
 
+    query.len = p - query.data;
```

Comment:
We should use `lua_assert()` to make sure the length is not exceeding the original upper limit `len`, just like the cosocket send code.